core/src/main/java/tech/tablesaw/io/xlsx/XlsxReadOptions.java

```diff
@@ -0,0 +1,44 @@
+package tech.tablesaw.io.xlsx;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.Reader;
+
+import tech.tablesaw.io.ReadOptions;
+
+public class XlsxReadOptions extends ReadOptions {
+
+    protected XlsxReadOptions(final Builder builder) {
+        super(builder);
+    }
+
+    public static Builder builder(File file) {
+        Builder builder = new Builder(file);
```

Comment:
can you write this as one line?

    return new Builder(file).tableName(file.getName());

Reply:
Actually you can just do `return new Builder(file)` because the super constructor already sets the table name based off the file name

Reply:
OK, but note that it doesn't work to return the value of the chained call to tableName, since its return type is the super-Builder-class, not this subclass. This is a weakness of the design, you need to introduce a type parameter **Builder<T extends Builder>** and use **return (T)** in the super class to allow this. That will make the superclass more complicated, but make the subclasses less.